analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -80,130 +107,88 @@ protected abstract class TokenClassifierBase
             };
 
             protected abstract SyntaxNode GetBindableParent(SyntaxToken token);
-            protected abstract bool IsDocComment(SyntaxTrivia trivia);
-            protected abstract bool IsRegularComment(SyntaxTrivia trivia);
             protected abstract bool IsKeyword(SyntaxToken token);
             protected abstract bool IsIdentifier(SyntaxToken token);
             protected abstract bool IsNumericLiteral(SyntaxToken token);
             protected abstract bool IsStringLiteral(SyntaxToken token);
 
-            protected TokenClassifierBase(SyntaxToken token, SemanticModel semanticModel, bool skipIdentifiers)
+            protected TokenClassifierBase(SemanticModel semanticModel, bool skipIdentifiers)
             {
-                this.token = token;
                 this.semanticModel = semanticModel;
                 this.skipIdentifiers = skipIdentifiers;
             }
 
-            public IEnumerable<TokenTypeInfo.Types.TokenInfo> Spans
-            {
-                get
-                {
-                    spans.Clear();
-                    ClassifyToken();
-
-                    foreach (var trivia in token.LeadingTrivia)
-                    {
-                        ClassifyTrivia(trivia);
-                    }
-
-                    foreach (var trivia in token.TrailingTrivia)
+            private TokenInfo TokenInfo(SyntaxToken token, TokenType tokenType) =>
+                string.IsNullOrWhiteSpace(token.ValueText)
+                    ? null
+                    : new()
                     {
-                        ClassifyTrivia(trivia);
-                    }
-
-                    return spans;
-                }
-            }
-
-            private void CollectClassified(TokenType tokenType, TextSpan span)
-            {
-                if (string.IsNullOrWhiteSpace(token.SyntaxTree.GetText().GetSubText(span).ToString()))
-                {
-                    return;
-                }
+                        TokenType = tokenType,
+                        TextRange = GetTextRange(token.GetLocation().GetLineSpan()),
+                    };
```

Comment:
Nitpick: Should be indented +4

Reply:
I tried this but if I do "Format Document" in VS the indentation is removed 😢.

Reply:
ah, ok